src/transformStateWithClones.js

```diff
@@ -7,7 +7,56 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateHistory = [];
+  let cloneState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(cloneState, action.extraData);
+        stateHistory.push({ ...cloneState });
```

Comment:
Consider making a push once after the switch statement

Reply:
![image](https://github.com/mate-academy/js_stateful-clones/assets/143609932/233c578c-194e-4c65-876a-305830108769)
like this?

Reply:
You can push once at the end of loop iteration.

Please ask all your questions in fe_chat  :)